FreeAPS/Sources/Services/ContactTrick/ContactTrickManager.swift

```diff
@@ -0,0 +1,196 @@
+import Algorithms
+import Combine
+import Contacts
+import Foundation
+import Swinject
+
+protocol ContactTrickManager {
+    func updateContacts(contacts: [ContactTrickEntry], completion: @escaping (Result<Void, Error>) -> Void)
+}
+
+final class BaseContactTrickManager: NSObject, ContactTrickManager, Injectable {
+    private let processQueue = DispatchQueue(label: "BaseContactTrickManager.processQueue")
+    private var state = ContactTrickState()
+    private let contactStore = CNContactStore()
+    private var workItem: DispatchWorkItem?
+
+    @Injected() private var broadcaster: Broadcaster!
+    @Injected() private var settingsManager: SettingsManager!
+    @Injected() private var apsManager: APSManager!
+    @Injected() private var storage: FileStorage!
+
+    private var contacts: [ContactTrickEntry] = []
+
+    let coreDataStorage = CoreDataStorage()
+
+    init(resolver: Resolver) {
+        super.init()
+        injectServices(resolver)
+
+        contacts = storage.retrieve(OpenAPS.Settings.contactTrick, as: [ContactTrickEntry].self)
+            ?? [ContactTrickEntry](from: OpenAPS.defaults(for: OpenAPS.Settings.contactTrick))
+            ?? []
+
+        broadcaster.register(GlucoseObserver.self, observer: self)
+        broadcaster.register(SuggestionObserver.self, observer: self)
+        broadcaster.register(SettingsObserver.self, observer: self)
+
+        configureState()
+    }
+
+    func updateContacts(contacts: [ContactTrickEntry], completion: @escaping (Result<Void, Error>) -> Void) {
+        processQueue.async {
+            self.contacts = contacts
+            self.renderContacts()
+            completion(.success(()))
+        }
+    }
+
+    private func configureState() {
+        processQueue.async {
+            let readings = self.coreDataStorage.fetchGlucose(interval: DateFilter().twoHours)
+            let glucoseValues = self.glucoseText(readings)
+            self.state.glucose = glucoseValues.glucose
```

Comment:
`self.state` doesn't have to be an instance field in this class - I could create a new one every time and pass it along into `renderContacts()` - but it was done like this in `WatchManager` so I kept it the same